lib/ws2/orders/cancel_by_gid.js

```diff
@@ -0,0 +1,23 @@
+'use strict'
+
+const Promise = require('bluebird')
+const debug = require('debug')('bfx:api:ws2:orders:cancel_by_gid')
+
+const cancelOrderByGid = (state = {}, opts) => {
+  const { ev, emit } = state
+  const { gid } = opts
+
+  debug('canceling order by gid: %s', gid)
+
+  emit('exec:order:cancel-by-gid', gid)
+
+  return new Promise((resolve, reject) => {
+    ev.once(`n:oc_multi-req:${gid}:success`, (notification, oc) => {
+      resolve(notification, oc)
+    })
+
+    ev.once(`n:oc_multi-req:${gid}:error`, reject)
+  })
+}
```

Comment:
memory-safe `watchResponse` from https://github.com/bitfinexcom/bfx-api-node-core/pull/23 should be used after it has been merged

Reply:
@robertkowalski FYI

Reply:
@avsek477 https://github.com/bitfinexcom/bfx-api-node-core/pull/23 is merged so you can sync your PR with master and update this code

Reply:
rebased to latest master and updated.